packages/react/src/products/hooks/types/useProductDetails.ts

```diff
@@ -0,0 +1,27 @@
+import type {
+  BreadCrumb,
+  Product,
+} from '@farfetch/blackout-client/src/products/types';
+import type { Error } from '@farfetch/blackout-client/src/types';
+import type {
+  GroupedEntriesAdapted,
+  ProductEntity,
+} from '@farfetch/blackout-redux/src/entities/types';
+import type { SizesAdapted } from '@farfetch/blackout-client/src/helpers/adapters/types';
+
+export type UseProductDetails = (id: number) => {
+  availableSizes: SizesAdapted;
+  breadcrumbs: BreadCrumb[] | undefined;
```

Comment:
What do you think?

```suggestion
  breadcrumbs?: BreadCrumb[];
```

Reply:
I prefer the `BreadCrumb[] | undefined` approach. Although the represent the same "data validity", in terms of presentation and developer experience they are different because the first approach forces the return object to have the `breadcrumbs` key to appear in the object and the other one does not :)